tests/test_connection.py

```diff
@@ -18,3 +28,17 @@ def test_response_content_type_handling(self, content_type, json, data):
         assert response.status_code == 200
         assert response.headers['Content-Type'] == content_type
         assert response.data == data
+
+    @mark.parametrize('headers', (
+        {}, {'app_name': 'name'}, {'app_id': 'id', 'app_key': 'key'}
+    ))
+    def test_request_with_headers(self, headers):
+        from bigchaindb_driver.connection import Connection
+        url = 'http://dummy'
+        connection = Connection(node_url=url, headers=headers)
+        with RequestsMock() as requests_mock:
+            requests_mock.add('GET', url, adding_headers=headers)
+            response = connection.request('GET')
+        assert response.status_code == 200
+        del response.headers['Content-type']
+        assert response.headers == headers
```

Comment:
Rather than doing a full compare, maybe it'd be safer to just check that the custom headers are in `response.headers`?